### PR TITLE
radius_circles

### PR DESCRIPTION
**Pull request by jannahhussain**
find out the area of the circle with the given radius

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,2 @@
-# python-projects
\ No newline at end of file
+# python-projects
+This is for Jannah to use as playground
```
Comment (lhussain) [inline README.md:2]: This line is not required

File: src/csv/area_and_circum_calculations.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+
+def do_build_data(current_position, line_array):
+    build_data = ""
+
+    if current_position > 1:
+        name = line_array[0]
+        radius = int(line_array[1])
+
+        a = " area = "
```
Comment (lhussain) [inline src/csv/area_and_circum_calculations.py]: Try not to use abbreviated variable naming like a.  Instead using real naming conventions e.g. area.

File: src/csv/area_and_circum_calculations.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        c = " circumference = "
```
Comment (lhussain) [inline src/csv/area_and_circum_calculations.py]: Try not to use abbreviated variable naming like c. Instead using real naming conventions e.g. circumference.

File: src/csv/area_and_circum_calculations.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import math
+
+
+def area_of_circles(radius):
```
Comment (lhussain) [inline src/csv/area_and_circum_calculations.py]: Function name here should not have been plural.  The behaviour of the function is to calculate area of a circle given the radius.

File: src/csv/area_and_circum_calculations.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    area = (math.pi * radius) * (math.pi * radius)
+    return area
+
+
+def circ_of_circles(radius):
```
Comment (lhussain) [inline src/csv/area_and_circum_calculations.py]: Function name here should not have been plural.  The behaviour of the function is to calculate circumference a circle given the radius.

File: src/csv/area_and_circ_of_circles.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+    current_position = 1
+
+    for line_array in csv_reader:
+        build_data = acc.do_build_data(line_array, current_position)
+
+        if build_data != "":
+            print(build_data)
+
+            current_position += 1
```
Comment (lhussain) [inline src/csv/area_and_circ_of_circles.py]: The indentation here is wrong, current position should always increment on each iteration of the loop.  However in your code the position only moves when the build_data is Not blank.  Position should always increment by one on every iteration of the loop.  So the current_position += 1 in line 16 has to be outside the if block.

**Review by lhussain — CHANGES_REQUESTED** [review]
Please read the comments and amend the code, test then re-commit changes

File: src/csv/area_and_circum_calculations.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import math
```
Comment (lhussain) [inline src/csv/area_and_circum_calculations.py:1]: Are you using math library anywhere?  If not then YAGNI

File: src/csv/area_and_circ_of_circles.py [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+import area_and_circum_calculations as acc
+
+
+with open('radius_of_circles.csv') as circles_file:
+    csv_reader = csv.reader(circles_file, delimiter=",")
+
+    current_position = 1
+
+    for line_array in csv_reader:
+        build_data = acc.do_build_data(line_array, current_position)
```
Comment (lhussain) [inline src/csv/area_and_circ_of_circles.py]: parameters are wrong way round

Comment (jannahhussain) [reply src/csv/area_and_circ_of_circles.py]: fixed

File: src/csv/area_and_circ_of_circles.py (hunk 6 above)
Comment (jannahhussain) [reply src/csv/area_and_circ_of_circles.py]: fixed

File: src/csv/area_and_circum_calculations.py (hunk 5 above)
Comment (jannahhussain) [reply src/csv/area_and_circum_calculations.py]: fixed

File: src/csv/area_and_circum_calculations.py (hunk 4 above)
Comment (jannahhussain) [reply src/csv/area_and_circum_calculations.py]: fixed

File: src/csv/area_and_circum_calculations.py (hunk 3 above)
Comment (jannahhussain) [reply src/csv/area_and_circum_calculations.py]: fixed

File: src/csv/area_and_circum_calculations.py (hunk 2 above)
Comment (jannahhussain) [reply src/csv/area_and_circum_calculations.py]: fixed

File: README.md (hunk 1 above)
Comment (jannahhussain) [reply README.md:2]: fixed

File: src/csv/area_and_circum_calculations.py [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+
+def area_of_circle(radius):
+    area = (math.pi * radius) * (math.pi * radius)
```
Comment (lhussain) [inline src/csv/area_and_circum_calculations.py]: This calculation is wrong.

Comment (jannahhussain) [reply src/csv/area_and_circum_calculations.py]: fixed

**Review by lhussain — APPROVED** [review]
Looks good to me

**Pull request merged**